 Patch to enhance the FBIOGET_FSCREENINFO ioctl,
 used by lcd_hack tool by BenH. (Sven Luther)
 added for kernel-source-2.6.10 (2.6.10-5)
 
 Status: Not submitted

diff -aurN a/drivers/video/aty/radeon_base.c b/drivers/video/aty/radeon_base.c
--- a/drivers/video/aty/radeon_base.c	2005-06-06 11:22:29.000000000 -0400
+++ b/drivers/video/aty/radeon_base.c	2005-06-15 21:55:56.000000000 -0400
@@ -1907,8 +1907,12 @@
 
 	fb_alloc_cmap(&info->cmap, 256, 0);
 
-	if (noaccel)
+	if (noaccel) {
 		info->flags |= FBINFO_HWACCEL_DISABLED;
+		info->fix.accel = FB_ACCEL_NONE;
+	} else {
+		info->fix.accel = FB_ACCEL_ATI_RADEON;
+	}
 
         return 0;
 }
